packages/dcos-history-service/extra/history/server.py

```diff
@@ -9,6 +9,37 @@
 app = Flask(__name__)
 compress = Compress()
 state_buffer = None
+log = logging.getLogger(__name__)
+add_headers_cb = None
+
+
+try:
+    import dcos_auth_python
+    log.info('dcos_auth_python module detected; applying settings')
+    add_headers_cb = dcos_auth_python.get_auth_headers
+except ImportError:
+    log.info('no dcos_auth_python module detected; using defaults')
+
+
+def headers_cb(add_headers_cb=add_headers_cb):
```

Comment:
why not just do a `global add_headers_cb` rather than doing this "Capture global as default argument" thing, which is a bit of odd corner behavior